### PR TITLE
bill-bug-fix-1

### PR DESCRIPTION
**Pull request by kachwalUttamsharma**
created new generic .js file for handling click outside modal, to close modal. which takes ref and callback function as argument.

open for suggestions.

### CODE REVIEW

**Comment by PremanandChowdhury** [comment]
@kachwalUttamsharma  
- Add the **GenericClosePopUp** Component to the other pop-ups too (like the profile image dropdown and filter in the homepage).

**Comment by kachwalUttamsharma** [comment]
> @kachwalUttamsharma
> 
> * Add the **GenericClosePopUp** Component to the other pop-ups too (like the profile image dropdown and filter in the homepage).

working on it

**Review by PremanandChowdhury — DISMISSED** [review]
Thanks @kachwalUttamsharma 🚀

**Comment by PremanandChowdhury** [comment]
@kachwalUttamsharma Resolve the conflicts. 👍🏼

**Comment by kachwalUttamsharma** [comment]
> @kachwalUttamsharma Resolve the conflicts. 👍🏼 

Ok

**Review by PremanandChowdhury — APPROVED** [review]

**Review by Kratika0907 — CHANGES_REQUESTED** [review]
Looks good. Can you please name the ref variable correctly. Everywhere it is const ref = ref.
We can have something like const billRef = ref.

**Comment by kachwalUttamsharma** [comment]
> an you please name the ref variable correctly. Everywhere it is const ref = ref.
> We can have something like const billRef = ref.

sure, working on it.

**Review by PremanandChowdhury — APPROVED** [review]
LGTM👍

**Review by Kratika0907 — APPROVED** [review]

File: components/filter/index.js [hunk 1]
```diff
@@ -1,35 +1,45 @@
...
+            Sort on credit
+          </li>
+          <li onClick={() => changeTransactions({ type: 'debit' })}>
+            Sort on debit
+          </li>
+          <li>Sort on date</li>
+          <li onClick={() => changeTransactions({ type: 'reset' })}>
+            Show original data
+          </li>
+        </div>
```
Comment (ankushdharkar) [inline components/filter/index.js:42]: <li> inside a <div>?

**Pull request merged**